app/services/service_result.rb

```diff
@@ -28,51 +28,68 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
+##
+# Represents the result of a call to a Service.
+#
+# @example
+#   result = Projects::UpdateService
+#     .new(user: current_user, model: @project)
+#     .call(permitted_params.project)
+#   result.success? # => true if the Service call was successful.
+#   result.result # => #<Project id: 1011>
+#   result.errors # => #<ActiveModel::Errors []>
+#
 class ServiceResult
   SUCCESS = true
   FAILURE = false
+  private_constant :SUCCESS, :FAILURE
+
+  # @return [Boolean] whether the Service call was successful.
+  attr_accessor :success
+
+  # Returns the result of the Service call. In typical usage this will be a model, i.e. an {ActiveRecord::Base}
+  # descendant.
+  #
+  # @return [Object, nil] the Service call result object. This can also be nil.
+  attr_accessor :result
+
+  # @return [ActiveModel::Errors] errors resulting from the Service call.
+  attr_accessor :errors
 
-  attr_accessor :success,
-                :result,
-                :errors,
-                :message,
-                :dependent_results
+  # @return [Array<ServiceResult>] all dependent ServiceResults - by virtue of the Service calling other services.
+  attr_accessor :dependent_results
 
-  attr_writer :state
+  attr_writer :message,
+              :state
 
   # Creates a successful ServiceResult.
-  def self.success(errors: nil,
-                   message: nil,
-                   message_type: nil,
-                   state: nil,
-                   dependent_results: [],
-                   result: nil)
-    new(success: SUCCESS,
-        errors:,
-        message:,
-        message_type:,
-        state:,
-        dependent_results:,
-        result:)
+  #
+  # @param (see #initialize)
+  # @return (see #initialize)
+  def self.success(**)
+    new(**, success: SUCCESS)
   end
 
+  ##
   # Creates a failed ServiceResult.
-  def self.failure(errors: nil,
-                   message: nil,
-                   message_type: nil,
-                   state: nil,
-                   dependent_results: [],
-                   result: nil)
-    new(success: FAILURE,
-        errors:,
-        message:,
-        message_type:,
-        state:,
-        dependent_results:,
-        result:)
+  #
+  # @param (see #initialize)
+  # @return (see #initialize)
+  def self.failure(**)
+    new(**, success: FAILURE)
   end
 
-  def initialize(success: false,
+  ##
+  # @api private
+  # @note Prefer using {.success} or {.failure} factory methods to calling {ServiceResult.new}.
```

Comment:
The `@api private` is not correct - after looking at `UseServiceResultFactoryMethods` cop, it looks like we are only discouraging use of `success: true/false` **keywords** in the constructor.

```
      #   # good
      #   ServiceResult.new(success: some_value)
```

is acceptable.